### ISSUE TITLE
Class not found b'org/jnius/NativeInvocationHandler'

### ISSUE THREAD

**Issue by Schnaffon** [kivy/pyjnius#223]
Hello,

I've been trying to use pyjnius lately, and find it rather useful.
However, I keep bumping into the same error : jnius.JavaException: Class
not found b'org/jnius/NativeInvocationHandler'

I started to wonder whether it was a problem in my code or in my installation.
To settle the question, I tried one of the examples provided on GitHub :
https://github.com/kivy/pyjnius/blob/master/tests/test_signature.py

And I got exactly the same error :

File "listjnius.py", line 119, in
a = TestImplem(*list(range(10)))
File "listjnius.py", line 68, in __init__
super(TestImplem, self).__init__(*args)
File "jnius/jnius_proxy.pxi", line 31, in
jnius.PythonJavaClass.__init__ (jnius/jnius.c:29526)
File "jnius/jnius_proxy.pxi", line 160, in jnius.create_proxy_instance
(jnius/jnius.c:31836)
File
"~/.virtualenvs/tests/lib/python3.5/site-packages/jnius/reflect.py",
line 154, in autoclass
c = find_javaclass(clsname)
File "jnius/jnius_export_func.pxi", line 25, in jnius.find_javaclass
(jnius/jnius.c:16682)
jnius.JavaException: Class not found b'org/jnius/NativeInvocationHandler'

This made me think the problem lies in my installation rather than in my code.

I installed Pyjnius from the github repo, so I'm using version 1.1.dev0.
I work on an ubuntu 16.04 recently updated, with python3.5 and java-8-openjdk-amd64.

I think I already saw a thread on this error here :
https://www.bountysource.com/issues/6892068-class-not-found-org-jnius-nativeinvocationhandler
But this thread is one year old, so the fix has already been merged, as
far as I know.

Do you have any insight regarding how it could be fixed ?

Thanks in advance,

Schnaffon

**Comment by cruor99** [comment]
I have been having that error as well.

What ended up working for me is using the particular branch mentioned above, https://github.com/VoiceThread/pyjnius.

The error does not happen using pyjnius normally, but if I expand the classpath as so:
`import jnius_config
jnius_config.expand_classpath()
jnius_config.add_classpath("libs/baxi/*")`

The error occurs.

I'm afriad I can't show more code, as it's under NDA from the provider :/

**Comment by dtorio** [comment]
Hi 
I have the same error:
jnius.JavaException: Class not found 'org/jnius/NativeInvocationHandler'
I'm using pyjnius to scan bluetooth LE beacons through a thread.
I'm using ubuntu 14.04, python2.7, and pyjnius 1.1.dev0
I guess this error can be resolved, using this commit [https://github.com/physion/pyjnius/commit/26f1367ab143cd72203f0db77556279eac8c304f](url)
Please merge it.
Thanks in advance.

**Comment by inclement** [comment]
@dtorio Your commit link seems to be broken, what commit is it?

**Comment by cruor99** [comment]
@inclement it seems to work if you just copy-paste it, rather than click it. [fixed markdown here](https://github.com/physion/pyjnius/commit/26f1367ab143cd72203f0db77556279eac8c304f)

**Comment by AndreMiras** [comment]
@cruor99 yes there's an issue/feature with JNI when calling FindClass() from a thread different than the main thread.
The good news is that the seems easy enough. I simply had to retrieve the `org.jnius.NativeInvocationHandler` manually in my main thread using:

    autoclass('org.jnius.NativeInvocationHandler')
That solved it for me.

**Comment by cruor99** [comment]
@AndreMiras If that works, I owe you a big one! I'll have to test it tomorrow.

**Comment by tomgold182** [comment]
@cruor99 Can you please share yor buildozer.spec file ?
I have tried to use the pyjnius fork you mentioned but it keeps on failing.

**Comment by cruor99** [comment]
@tomgold182 I never ran into this issue on android itself - only when using it on the desktop with a different threaded library. I'm honestly not even sure if this is still an issue for me.

I'll do some tests and get back to you if I can, but to reiterate: I never had an issue with pyjnius when run on android.

**Comment by tomgold182** [comment]
@cruor99  thanks!
it appears when trying to use plyer's camera.take_picture function

**Comment by xloem** [comment]
It looks like https://github.com/physion/pyjnius/commit/26f1367ab143cd72203f0db77556279eac8c304f makes other changes to the codebase too.  Could anybody isolate just the change that helps with some instances of this issue, and present it as a pull request?  Any idea why it isn't a pull request already, or if it is one which one it is?
